timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/AwaitedLocksCollection.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.lock;
+
+import java.util.UUID;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Supplier;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Maps;
+
+// TODO(nziebart): can we combine this logic with HeldLocksCollection?
+// Or, should awaitLocks just be implemented by a lock + unlock?
```

Comment:
We probably can - though I think it's perfectly fine to have this class which encapsulates the idea of 'lock requests that are still in flight'.

Lock + unlock is simple, though it would be nice if we could benchmark the overheads involved since it is a stronger synchronization than what we need.